pymatgen/io/openmm/setup.py

```diff
@@ -0,0 +1,225 @@
+from abc import ABC
+
+import openmm
+import openmm.app
+from openbabel import pybel
+from openff.toolkit.typing.engines import smirnoff
+from openmm.app import Simulation
+from pymatgen.io.babel import BabelMolAdaptor
+
+from openmm.app.topology import Topology
+from openmm import LangevinMiddleIntegrator
+
+from pymatgen.io.packmol import PackmolBoxGen
+from pymatgen.io.xyz import XYZ
+
+from openmm.unit import *
+
+import numpy as np
+import parmed
+import openff
+
+import pathlib
+import tempfile
+from typing import Dict, Union, Optional, Tuple
+
+
+# does this belong as an attribute of the Molecule class?
+def smile_to_mol(smile):
+    """
+    Converts a SMILE to a Pymatgen Molecule.
+
+    Parameters
+    ----------
+    smile: a SMILE.
+
+    Returns
+    -------
+    Pymatgen.Molecule
+    """
+    mol = pybel.readstring("smi", smile)
+    mol.addh()
+    mol.make3D()
+    adaptor = BabelMolAdaptor(mol.OBMol)
+    return adaptor.pymatgen_mol
+
+# TODO: this logic all belongs in inputs/OpenMMGenerator
+
+class OpenMMSimulationGenerator:
+    """
+    An opinionated generator for OpenMM Simulations sets.
```

Comment:
What does "opinionated" mean in this context? :)  I'm not familiar.

Reply:
I suppose I wanted to imply that it does not implement the full OpenMM feature set but instead makes it easier to setup a subset of OpenMM simulations. That language is probably not super clear and redudant with the general Pymatgen approach.

Regardless, I am eventually going to transfer all content in `setup.py` to `inputs.py`. The `setup.py` code was my first pass but is pretty messy and doesn't use the new abstract IO.